src/ft2font_wrapper.cpp

```diff
@@ -664,7 +668,9 @@ static PyObject *PyFT2Font_set_text(PyFT2Font *self, PyObject *args, PyObject *k
         return NULL;
     }
 
-    CALL_CPP("set_text", self->x->set_text(size, &codepoints[0], angle, flags, xys));
+    if (size > 0) {
+        CALL_CPP("set_text", self->x->set_text(size, &codepoints[0], angle, flags, xys));
```

Comment:
I don't know how much we can rely on C++11 -- we'd have to test it.  I think gcc is fine, but clang and MSVC may be playing catch up still (not sure).  

Anyway, your workaround seems like a good suggestion.  I suspect what's happening here is that no buffer is allocated at all in the zero-length case, so `&codepoints[0]` is garbage.